File: Tree/SimpleTree.py
class SimpleTree:
    def __init__(self):
        self.root = None
        self.count=0
        
    class Node:
        def __init__(self,data):
            self.ele = data
            self.left= None
            self.right= None
    
    def isEmpty(self):
        return self.count==0
    
    def getCount(self):
        return self.count

    def insert(self,data):
        current = par_naode = self.root
        while current != None and current.ele!=data:
            par_naode = current
            if data < current.ele:
                current = current.left
            else:
                current = current.right
            
        if current == None:
            node = self.Node(data)
            if par_naode !=None:
                if data<par_naode.ele:
                    par_naode.left = node
                else:
                    par_naode.right = node
            else:
                self.root = node
            self.count+=1

    def travers(self):
        print('pretorder')
        self.preorder(self.root)
        print('inorder')
        self.inorder(self.root)
        print('postorder')
        self.postorder(self.root)

    def inorder(self,node):
        if node!=None:
            self.inorder(node.left)
            print(node.ele)
            self.inorder(node.right)
    
    def postorder(self,node):
        if node!=None:
            self.postorder(node.left)
            self.postorder(node.right)
            print(node.ele)
    
    def preorder(self,node):
        if node!=None:
            self.preorder(node.left)
            self.preorder(node.right)
            print(node.ele)

    def isMember(self,data):
        node = self.root
        while node != None:
            if data == node.ele:
                return True
            if data<node.ele:
                node = node.left
            else:
                node = node.right
        return False
    
    def deleteNode(self,key):
        if not self.isEmpty():
            self.root = self._deleteNode_(self.root,key)
    
    def _deleteNode_(self,node,key):
        if node is None:
            return None
    
        if node.ele>key:
            node.left = self._deleteNode_(node.left,key)
        elif node.ele<key:
            node.right = self._deleteNode_(node.right,key)
        else:
            if node.left is None:
                return node.right
            elif node.right is None:
                return node.left
            else:
                succ = self.getMin(node.right)
                node.ele = succ.ele
                node.right = self._deleteNode_(node.right,succ.ele)
        return node

    def getMin(self,node):
        current = node
        while current.left is not None:
            current = current.left 
        return current

File: java/Tree/simpleTree.java
public class simpleTree {
    Node root = null;
    int count = 0;

    simpleTree() {

    }

    class Node {
        int data;
        Node left = null;
        Node right = null;

        public Node(int ele) {
            data = ele;
        }
    }

    public void insert(int ele) {
        root =_insert(ele, root);
    }
    
    private Node _insert(int ele, Node node) {
        if (node == null) {
            return new Node(ele);
        }
        if (ele < node.data) {
            node.left = _insert(ele, node.left);
        } else if (ele > node.data) {
            node.right = _insert(ele, node.right);
        }
        return node;
    }
    
    public void delete(int key){
        _delete_(root, key);
    }

    private Node _delete_(Node node , int key){
        if (node == null){
            return node;
        }

        if(node.data>key){
            node.left = _delete_(node.left, key);
        } else if(node.data<key){
            node.right = _delete_(node.right, key);
        }
        else{
            if(node.left == null){
                return node.right;
            }else if(node.right == null){
                return node.left;
            }else{
                Node succ = getMin(node.right);
                node.data = succ.data;
                node.right = _delete_(node.right, succ.data);
            }
        }
        return node;
    }

    private Node getMin(Node node){
        Node temp = node;
        while (temp.left != null) {
            temp = temp.left;
        }
        return temp;
    }

    public void inorder() {
        _inorder(root);
        System.out.println();
    }

    private void _inorder( Node node) {
        if (node!=null){
            _inorder(node.left);
            System.out.print(node.data+" ");
            _inorder(node.right);
        }
    }

    public void preorder() {
        _preorder(root);
        System.out.println();
    }

    private void _preorder( Node node) {
        if (node!=null){
            System.out.print(node.data+" ");
            _preorder(node.left);
            _preorder(node.right);
        }
    }

    public void postorder() {
        _postorder(root);
        System.out.println();
    }

    private void _postorder( Node node) {
        if (node!=null){
            _postorder(node.left);
            _postorder(node.right);
            System.out.print(node.data+" ");
        }
    }
}